src/storage/patch/ContainerPatcher.ts

```diff
@@ -0,0 +1,50 @@
+import { BasicRepresentation } from '../../ldp/representation/BasicRepresentation';
+import type { Representation } from '../../ldp/representation/Representation';
+import { INTERNAL_QUADS } from '../../util/ContentTypes';
+import { NotImplementedHttpError } from '../../util/errors/NotImplementedHttpError';
+import { isContainerIdentifier } from '../../util/PathUtil';
+import { SOLID_META } from '../../util/Vocabularies';
+import type { RepresentationPatcherInput } from './RepresentationPatcher';
+import { RepresentationPatcher } from './RepresentationPatcher';
+
+/**
+ * A `RepresentationPatcher` specifically for patching containers.
+ * A new body will be constructed from the metadata by removing all generated metadata.
+ * This body will be passed to the wrapped patcher.
+ */
+export class ContainerPatcher extends RepresentationPatcher {
+  private readonly patcher: RepresentationPatcher;
+
+  public constructor(patcher: RepresentationPatcher) {
+    super();
+    this.patcher = patcher;
+  }
+
+  public async canHandle(input: RepresentationPatcherInput): Promise<void> {
+    const { identifier, representation } = input;
+    if (!isContainerIdentifier(identifier)) {
+      throw new NotImplementedHttpError('Only containers are supported.');
+    }
+    let dummyRepresentation = representation;
+    if (representation) {
+      dummyRepresentation = new BasicRepresentation([], representation.metadata, INTERNAL_QUADS);
+    }
+    await this.patcher.canHandle({ ...input, representation: dummyRepresentation });
+  }
+
+  public async handle(input: RepresentationPatcherInput): Promise<Representation> {
+    const { identifier, representation } = input;
+    if (!representation) {
+      return await this.patcher.handle(input);
+    }
+    // Remove all generated metadata to prevent it from being stored permanently
```

Comment:
There is a chance yes (also for the converting patcher since we only "simulate" the conversion in the canHandle there).

You can just say that you really want to use a `WeakMap` though 😉. We could do that yes, but would require copying the metadata (since currently we modify it in place). It does feel a bit weird to shift the creation of the new representation to the canHandle function.